### PR TITLE
Stop using django-annoying

### LINKED ISSUE #40: Stop using django-annoying
It's more annoying to have too many dependencies.

### PR DESCRIPTION
**Pull request by sevein**
This closes #40.

### CODE REVIEW

**Review by jhsimpson — COMMENTED** [review]
I haven't tested this but from reading the code it looks good to me.  I think this is a good idea to merge.

File: fpr/tests.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            'format': fprule.format.uuid,
+            'command': fprule.command.uuid,
+        })
+        self.assertFalse(form.is_valid())
+        self.assertEquals(form.non_field_errors(), [
+            'An identical FP rule already exists.'
+            ' See rule {}.'.format(fprule.uuid)
+        ])
+
+    def create_fprule(self):
```
Comment (jrwdunham) [inline fpr/tests.py]: Nitpick: could be a func or staticmethod cuz self ain't used.

File: fpr/tests.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import uuid
```
Comment (jrwdunham) [inline fpr/tests.py]: Unused import. (Is flake8 not being run on this?)

File: fpr/forms.py [hunk 3]
```diff
@@ -50,21 +49,12 @@ class Meta:
...
-            )
-
-            if existing_idtool:
-                raise forms.ValidationError(
-                    _('An ID tool with this description and version already'
-                      ' exists'))
-
+        if self.instance.pk is None and fprmodels.IDTool.objects.filter(
+            description=cleaned_data.get('description'),
+            version=cleaned_data.get('version')).exists():
```
Comment (jrwdunham) [inline fpr/forms.py]: Nitpick: I think you should add 4 spaces indentation before lines 53 and 54 (and pylint agrees with me).

File: fpr/tests.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            uuid='37f3bd7c-bb24-4899-b7c4-785ff1c764ac')
+        cmd2 = IDCommand(description='Foobar')
+        cmd2.save(replacing=cmd1)
+
+        descendants = get_revision_descendants(IDCommand, cmd1.uuid, [123])
+        self.assertEqual(descendants, [123, cmd2])
+
+        with self.assertRaises(IDCommand.DoesNotExist):
+            descendants = get_revision_descendants(
+                IDCommand, '12345', [])
```
Comment (jrwdunham) [inline fpr/tests.py]: Can you provide more detail on how to run these tests? I was unable to do so easily.

Also, looking at the definition of `get_revision_descendants` it looks like that func will not raise the exception that you are expecting to catch here. It looks like it catches that exception and just passes. 

Relatedly, I noticed that travis is not running these tests. It only runs those in testproject/. If this were fixed, then we could see if these tests were passing.

File: fpr/tests.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        resp = self.client.get(url)
+        self.assertEqual(resp.context['form'].initial['tool'], None)
+
+        resp = self.client.get(url, {'parent': 12345})
+        self.assertEqual(resp.context['form'].initial['tool'], None)
+
+        resp = self.client.get(url, {'parent': tool.uuid})
+        self.assertEqual(resp.context['form'].initial['tool'], tool)
+
+    def test_fpommand_create(self):
```
Comment (jrwdunham) [inline fpr/tests.py]: Typo: `fpommand`

**Review by jrwdunham — CHANGES_REQUESTED** [review]
I'd like to get the tests introduced here running in Travis as well as have tox or some docs explaining how to run them.

**Comment by sevein** [comment]
Thanks @jrwdunham. Can you take another look?

`get_revision_descendants` raises an exception when the object cannot be found. We ignore exceptions when looking up descendants though.

**Review by jrwdunham — APPROVED** [review]
> get_revision_descendants raises an exception when the object cannot be found. We ignore exceptions when looking up descendants though.

Right. Oops. LGTM. 👍

**Comment by sevein** [comment]
a62404e48118ed54c9c26457663bbbbd19f32aaf
a762136e17e8a1b981db691e21e6292557a63baa
333af7b65fd9054027d28815534dc0966831e202

**Pull request merged**